CMakeLists.txt

```diff
@@ -1,96 +1,141 @@
+find_package(Eigen3 REQUIRED)
+find_package(real_time_tools REQUIRED)
+find_package(time_series REQUIRED)
+find_package(signal_handler REQUIRED)
+find_package(serialization_utils REQUIRED)
+find_package(Threads)
+find_package(rt)
+
+# export de dependencies
+ament_export_dependencies(pybind11 Eigen3 real_time_tools time_series
+                          signal_handler serialization_utils)
+
+# prepare to export all needed targets
+set(all_targets)
+set(all_target_exports)
+
+#
+# Add main target.
+#
+add_library(${PROJECT_NAME} INTERFACE)
+# Add the include dependencies
+target_include_directories(
+  ${PROJECT_NAME}
+  INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+            $<INSTALL_INTERFACE:include>)
+# Link the dependencies
+target_link_libraries(${PROJECT_NAME} INTERFACE rt::rt)
+target_link_libraries(${PROJECT_NAME} INTERFACE Eigen3::Eigen)
+target_link_libraries(${PROJECT_NAME} INTERFACE real_time_tools::real_time_tools)
+target_link_libraries(${PROJECT_NAME} INTERFACE time_series::time_series)
+target_link_libraries(${PROJECT_NAME} INTERFACE signal_handler::signal_handler)
+target_link_libraries(${PROJECT_NAME} INTERFACE serialization_utils::serialization_utils)
+target_link_libraries(${PROJECT_NAME} INTERFACE Threads::Threads)
+target_link_libraries(${PROJECT_NAME} INTERFACE pybind11::embed)
+# Export the target.
+ament_export_interfaces(export_${PROJECT_NAME} HAS_LIBRARY_TARGET)
+list(APPEND all_targets ${PROJECT_NAME})
+list(APPEND all_target_exports export_${PROJECT_NAME})
+
+#
+# manage the demos.
+#
 add_executable(demo demos/demo.cpp)
-target_link_libraries(demo ${catkin_LIBRARIES})
+target_link_libraries(demo ${PROJECT_NAME})
+list(APPEND all_targets demo)
 
 add_executable(demo_multiprocess_backend demos/demo_multiprocess_backend.cpp)
-target_link_libraries(demo_multiprocess_backend ${catkin_LIBRARIES}
-    rt pthread)
+target_link_libraries(demo_multiprocess_backend ${PROJECT_NAME})
+list(APPEND all_targets demo_multiprocess_backend)
 
 add_executable(demo_multiprocess_frontend demos/demo_multiprocess_frontend.cpp)
-target_link_libraries(demo_multiprocess_frontend ${catkin_LIBRARIES}
-    rt pthread)
-
-#########################
-# manage the unit tests #
-#########################
-add_subdirectory(tests)
-
-###################
-# python bindings #
-###################
-
-catkin_add_pybind11_module(py_generic)
-catkin_add_pybind11_module(py_finger_types)
-catkin_add_pybind11_module(py_trifinger_types)
-catkin_add_pybind11_module(py_one_joint_types)
-catkin_add_pybind11_module(py_two_joint_types)
-
+target_link_libraries(demo_multiprocess_frontend ${PROJECT_NAME})
+list(APPEND all_targets demo_multiprocess_frontend)
+
+#
+# manage the unit tests.
+#
+if(BUILD_TESTING)
+  add_subdirectory(tests)
+endif()
 
-##########################
-# building documentation #
-##########################
-add_doxygen_documentation()
+#
+# python bindings.
+#
+
+macro(ADD_PYTHON_BINDINGS module_name)
```

Comment:
@MaximilienNaveau What is the reason for not using `pybind11_add_module` anymore?
While porting other packages, I was using the following (based on what we had for catkin, I only needed to update the install destination) which works for me and is much shorter:

```cmake
macro(add_pybind11_module module_name)
    pybind11_add_module(${module_name} srcpy/${module_name}.cpp)
    target_include_directories(${module_name} PUBLIC include)
    target_link_libraries(${module_name} PRIVATE ${ARGN})
    install(TARGETS ${module_name}
        DESTINATION "${PYTHON_INSTALL_DIR}/${PROJECT_NAME}")
endmacro()
```

The install destination below should be changed to `"${PYTHON_INSTALL_DIR}/${PROJECT_NAME}"` for the targets to end up at the right place (right now imports like `import robot_interfaces.py_generic` are not working anymore).

I can fix this as I'm anyway working on migrating my other packages.  I'm wondering, though, if there is some problem with the shorter macro, which I am not aware of.

Another question:  As we don't have pybind11_catkin anymore (which is where the above macro was defined), where should I put this now?  Would it fit into mpi_cmake_modules or should I rather create a new package for it?